packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -196,9 +196,6 @@ def _get_create_batch_tx(
         """Create a batch of items."""
         # create the items
         self.nonce += 1
-        nonce = ledger_api.api.eth.getTransactionCount(deployer_address)
-        if self.nonce != nonce:
-            self.nonce = nonce
```

Comment:
self.nonce is your off-chain reference to the nonce, which is generally ahead of the ledger api nonce. Hence, I would just check that `nonce <= self.nonce`, the assignment makes no sense in my opinion.